### PR TITLE
Add CurrencyDesign document.

### PR DESCRIPTION
**Pull request by rahulgi**
I wanted to have a definitive document upon which we can iterate. This currently lacks details about trust levels and adds some stuff which we may not want, but I think it'll be good for us to have something concrete. Also, is Github the right place to host this kind of document? I like that it's public and allows for discussion, contribution, and comment, but it's not necessarily the best place for reading.

### CODE REVIEW

File: CurrencyDesign.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Raha Currency Design Document
+
+## Summary
+
+Raha is a digital currency built around verified identities and implementing an inheritance-funded universal basic income. The goal of Raha is to provide a sustainable basic income to everybody in the world.
+
+Raha will eventually be implemented as a distributed, cryptographically secured and trusted digital currency. Today, it is implemented as a centrally-controlled digital currency backed by a blockchain-based tamper-evident audit log.
```
Comment (markulrich) [inline CurrencyDesign.md]: "Today, it is implemented as a centrally controlled restricted-write blockchain backed by a publicly readable, tamper-evident, decentralized audit log"

File: CurrencyDesign.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+In the future, Raha may create "savings" accounts for identities which are not considered inactive as quickly.
+
+Raha members can also donate their income back to the network if they do not need it.
+
+#### Progressively increasing inactivity period
+
+As an incentive to increase transaction volume, Raha may implement a much shorter and progressively increasing inactivity period, either graded by time or transaction volume.
+
+For example, for the first year of Raha's existence, funds may be reclaimed after only 30 days of identity activity. In the second year, that period may be extended to 6 months. In the third year, the full 1 year burden may be implemented.
```
Comment (markulrich) [inline CurrencyDesign.md]: Cut?

File: CurrencyDesign.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+Raha will initially create 10 billion coins. These will be distributed in increments of 5 million coins per day, divided amongst all active members of the network.
+
+### Monetary policy
+
+#### The 20 percent rule
+
+20% of the initial distribution and reclaimed moneys will be given to the Raha Foundation. This will go to support the network development, growth, and maintenance.
+
+#### Reserve-backed currency
```
Comment (markulrich) [inline CurrencyDesign.md]: Cut?

File: CurrencyDesign.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+The Raha Foundation may seek out donations from individual and corporate sponsors to fund its operations and reserves. We see Raha as an especially good partner for companies who hope to offset the human impact of automating products and processes.
+
+### Applications and marketplaces
+
+A diverse ecosystem of applications and marketplaces is necessary to create a stable foundation of demand for Raha. Some potential applications are listed below:
+
+#### Goods and services marketplace
+
+Simple and self explanatory.
```
Comment (markulrich) [inline CurrencyDesign.md]: ?

File: CurrencyDesign.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+#### Direct messaging
+
+Raha's network of verified identities is an extremely valuable channel for organizations looking to reach real people. Raha members could opt-in to being contacted with, for example, push notifications from advertisers, in exchange for Raha. Members could set the price of contact based on their profile and depth of verification, creating a market around highly-verified users.
+
+#### Quantified human marketplace
+
+Raha's network of verified identities provides an extremely valuable research and machine-learning data resource. Raha members could opt-in to contributing their data to aggreggated data sets in exchange for compensation in Raha. Eventually, it could be a point of pride for organizations to use fairly-compensated data sourced from willing Raha members.
```
Comment (markulrich) [inline CurrencyDesign.md]: Awesome!

File: CurrencyDesign.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+Development and maintenance of the Raha technology and ecosystem are managed by the Raha Foundation, a nonprofit organization dedicated to achieving Raha's goal of providing a sustainable basic income to everbody in the world. Eventually, the Raha Foundation will be itself be governed by a board of directors elected by all Raha members via a system of liquid democracy. Today, the Raha Foundation is controlled by its four founding members, Mark Ulrich, Tina Roh, Omar Diab, and Rahul Gupta-Iwasaki.
+
+This document is the source-of-truth for describing the Raha currency and economic model. It lays out network structure, currency policy, and metrics for success. This is a living document, and it will be updated with changes to any of the above as they are made. All changes will be made transparently and with open dialogue preceding them.
+
+## Values
+
+Raha is founded upon the ideals that people are born with the fundamental right to a baseline standard of living. 
+
+## Objective
```
Comment (markulrich) [inline CurrencyDesign.md]: Cut?

File: CurrencyDesign.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+In a society with high levels of wealth inequality, a basic income is redistributive. By 'sustainable' basic income, Raha seeks to attain a balance between establishing a baseline level of well-being and retaining the free market's ability to incentivize productivity. The target income level, `u`, shall be periodically reset by the Raha Foundation to a reasonable value given the global economy's current levels of productivity.
+
+Today: `u = $10/month`.
+
+Raha will not seek to provide a basic income beyond `u`.
+
+The value of `x` is determined by the value of the Raha currency, which we shall call `z`. The value of `z` will increase in proportion to demand for the Raha currency. Demand for Raha currency can be measured by the number, `a`, and size, `b`, of transactions taking place on the network.
+
+### Key metrics
```
Comment (markulrich) [inline CurrencyDesign.md]: Cut?

File: CurrencyDesign.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+Raha will eventually be implemented as a distributed, cryptographically secured and trusted digital currency. Today, it is implemented as a centrally-controlled digital currency backed by a blockchain-based tamper-evident audit log.
+
+Development and maintenance of the Raha technology and ecosystem are managed by the Raha Foundation, a nonprofit organization dedicated to achieving Raha's goal of providing a sustainable basic income to everbody in the world. Eventually, the Raha Foundation will be itself be governed by a board of directors elected by all Raha members via a system of liquid democracy. Today, the Raha Foundation is controlled by its four founding members, Mark Ulrich, Tina Roh, Omar Diab, and Rahul Gupta-Iwasaki.
+
+This document is the source-of-truth for describing the Raha currency and economic model. It lays out network structure, currency policy, and metrics for success. This is a living document, and it will be updated with changes to any of the above as they are made. All changes will be made transparently and with open dialogue preceding them.
```
Comment (markulrich) [inline CurrencyDesign.md]: Maybe just `"This document lays out network structure, currency policy, and metrics for success."` for now ... sources of truth belong in the blockchain one day ;)

File: CurrencyDesign.md (hunk 7 above)
Comment (rahulgi) [reply CurrencyDesign.md]: Are you suggesting cutting the objective and key metrics sections? I think it's really important to define our goal and progress towards that in a quantifiable way.

File: CurrencyDesign.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+## Values
+
+Raha is founded upon the ideals that people are born with the fundamental right to a baseline standard of living.
+
+## Objective
+
+Raha's goal is to provide a sustainable basic income to everybody in the world. This can be quantified with the following utility function:
+
+`utility = n * (1 + log(1 / u + x))`
```
Comment (osdiab) [inline CurrencyDesign.md]: So is what you're saying that we're trying to maximize this utility function? I'm a little confused what exactly you're trying to express here, and I also think that this mathematical approach might be appealing to some but doesn't capture what we're qualitatively trying to achieve. Much rather have this more as a footnote than the first, primary way we express our objective.

File: CurrencyDesign.md (hunk 7 above)
Comment (osdiab) [reply CurrencyDesign.md]: I'm fine with it being there as an aside on how we're measuring our progress or under some different label, but it's not a compelling story for what we value and why we're doing this. It's a low appeal, confusing to most way of expressing what our goals are, and the actual proportions of importance and stuff are largely arbitrary—for instance, logarithms generally scale so far below linear that we could probably maximize this function by largely ignoring the size of the UBI altogether and just getting as many people on the network as we can, like by putting some arbitrary game or social networking element on the platform.

File: CurrencyDesign.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+A key part of the value of a basic income comes from people's ability to depend on it over time. A predictable, stable basic income enables people to take risks and pursue opportunities without fear of falling below the baseline standard of living that the income guarantees.
+
+Relying too much on any single source of demand for a currency makes the value of that currency highly susceptible to fluctuations in that source of demand. A secondary objective of Raha is to cultivate not just demand but a diverse array of applications and marketplaces that drive that demand to ensure a stable price foundation.
+
+## Currency design
+
+Raha will eventually be implemented as a distributed, cryptographically secured and trusted digital currency. Today, it is implemented as a centrally-controlled restricted-write blockchain backed by a publicly readable, tamper-evident, decentralized audit log.
+
+As a digitial currency, using Raha for everday transactions should be as easy as or easier than traditional paper currencies. At a system level, Raha's core differentiating features are:
```
Comment (osdiab) [inline CurrencyDesign.md]: The system level points seem to be totally irrelevant than the ease of use point. Odd that they're mashed together in a sentence

File: CurrencyDesign.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+
+An identity on the Raha network which has not been verified by anybody is essentially an anonymous public key. The Raha UBI will only be distributed to those identities which have been verified by a highly trusted verifier.
+
+Initially, the Raha network supports verification via video-based social trust coupled with Google, Facebook, or Github authentication. A more extensive list of current and potential verifiers is below:
+
+#### Modular identity verifiers
+
+* Video-based social trust: New users create a video with an existing user and then have their account "trusted" by that existing user.
+* Google, Facebook, or Github authentication: self explanatory. In conjunction with video-based social trust constitutes verification for UBI distribution.
+* Passport: UBI-C is implementing identity verification based off of the RFID chips present in passports. This is an interesting verifier that we could also implement.
```
Comment (osdiab) [inline CurrencyDesign.md]: May mention what UBI-C is, or just introduce the concept and leave a footnote saying that the idea has been posited by them

File: CurrencyDesign.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+* Other Government identity schemes like Adhar.
```
Comment (osdiab) [inline CurrencyDesign.md]: Aadhar

File: CurrencyDesign.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+Raha supports a modular system of identity verification. The Foundation will maintain a list of trusted verifiers including government-provided identifiers, social networks, and identity applications which can then be used to verify a given identity. As each verifier understands different parts of a person's identity, applications which consume Raha identities can then themselves decide which verifiers to trust.
+
+An identity on the Raha network which has not been verified by anybody is essentially an anonymous public key. The Raha UBI will only be distributed to those identities which have been verified by a highly trusted verifier.
+
+Initially, the Raha network supports verification via video-based social trust coupled with Google, Facebook, or Github authentication. A more extensive list of current and potential verifiers is below:
+
+#### Modular identity verifiers
+
+* Video-based social trust: New users create a video with an existing user and then have their account "trusted" by that existing user.
+* Google, Facebook, or Github authentication: self explanatory. In conjunction with video-based social trust constitutes verification for UBI distribution.
```
Comment (osdiab) [inline CurrencyDesign.md]: More generally, Federated Identity providers, which we can split into centralized (Google, Facebook, Github) or decentralized (Civic, uPort, etc).

Though for each of the centralized ones, we should probably make it clear that just having an account isn't sufficient—having accounts with lots of connections, whose behavior suggests that they aren't corresponding to fake identities is necessary

File: CurrencyDesign.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+
+Raha members can also donate their income back to the network if they do not need it.
+
+#### Progressively increasing inactivity period
+
+As an incentive to increase transaction volume, Raha may implement a much shorter and progressively increasing inactivity period, either graded by time or transaction volume.
+
+### Initial distribution
+
+Raha will initially create 10 billion coins. These will be distributed in increments of 5 million coins per day, divided amongst all active members of the network.
```
Comment (osdiab) [inline CurrencyDesign.md]: So the amount distributed per cycle is set, ie the basic income isn't set based on num coins per user?

Also, is a daily trickle necessarily the dynamics we want, or is there some other better distribution policy? there was a really interesting podcast episode on lottery style savings accounts, how they got a ton more people to save instead of buying lottery tickets by reframing savings as a regular lottery payout instead of a regular, small percentage interest payout—most states made them illegal because they competed against the government-run lotteries, but even if we're not doing something like that it highlights that the nature of how it's distributed has a strong impact on how the UBI appeals to people.

File: CurrencyDesign.md [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+
+### Inheritance-funded basic income
+
+All verified identities in the Raha network receive a basic income. This basic income is funded by periodically reclaiming the Raha held by deceased/inactive users and redistributing it to the network in what is essentially a 100% estate tax on wealth in Raha.
+
+There is no mechanism to prevent wealth from being transferred to another identity or out of Raha prior to reclamation, but we hope to achieve a cultural shift where people give back to society as much as their progeny. An estimated 4 trillion dollars is inherited globally every year. Divided evenly between the estimated 8 billion people on Earth, that inheritance could fund a UBI of 5 hundred dollars per person per year.
+
+An identity will be marked deceased upon provision of proof of death. However, as most identities will likely never have that proof provided, an identity which is inactive for a period of 1 year will also have its funds reclaimed.
+
+In the future, Raha may create "savings" accounts for identities which are not considered inactive as quickly.
```
Comment (osdiab) [inline CurrencyDesign.md]: What's the reason for that? Seems arbitrary

File: CurrencyDesign.md [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+
+Raha is not in the business of re-inventing identity verification. We are happy to trust the efforts of others here.
+
+#### Technical implementation
+
+The Foundation signs an operation trusting the identity verifier. The verifier then signs an operation verifying an identity. Both operations can have optional expiry dates, and untrust/verify operations can also occur.
+
+### Inheritance-funded basic income
+
+All verified identities in the Raha network receive a basic income. This basic income is funded by periodically reclaiming the Raha held by deceased/inactive users and redistributing it to the network in what is essentially a 100% estate tax on wealth in Raha.
```
Comment (osdiab) [inline CurrencyDesign.md]: Should we mention that we would consider having some way of naming benificiaries in the future?

File: CurrencyDesign.md [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+
+#### Progressively increasing inactivity period
+
+As an incentive to increase transaction volume, Raha may implement a much shorter and progressively increasing inactivity period, either graded by time or transaction volume.
+
+### Initial distribution
+
+Raha will initially create 10 billion coins. These will be distributed in increments of 5 million coins per day, divided amongst all active members of the network.
+
+### Monetary policy
```
Comment (osdiab) [inline CurrencyDesign.md]: Are we not considering any inflationary system of minting coins to distribute? I think we agreed that having an excessively high rate of inflation is problematic, but that doesn't preclude having none at all.

File: CurrencyDesign.md [hunk 18]
```diff
@@ -0,0 +1,150 @@
+# Raha Currency Design Document
```
Comment (osdiab) [inline CurrencyDesign.md]: Might make sense to put timestamps in the filename to enforce a non-alphabetical ordering for the documents it contains.

Also, "documentation" makes it sound like documentation for code. We should choose a different word if this is a separate repo. Like "specs" or something?

It is also questionable if a github document is really the most democratic way to host this, because in all likelihood the only people that would look on it in github form are coders, who are exactly the demographic of people who don't need to worry about nice UIs in apps, or losing cryptographic keys. If anything, public Google Drive docs with public commenting probably makes more sense

File: CurrencyDesign.md [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+
+As an incentive to increase transaction volume, Raha may implement a much shorter and progressively increasing inactivity period, either graded by time or transaction volume.
+
+### Initial distribution
+
+Raha will initially create 10 billion coins. These will be distributed in increments of 5 million coins per day, divided amongst all active members of the network.
+
+### Monetary policy
+
+#### The 20 percent rule
```
Comment (osdiab) [inline CurrencyDesign.md]: Phrasing it as "the 20 percent rule" makes it sound like it's not going to be adaptive to the relative value of the currency nor the actual financial needs of maintainers, but is just this undying truth. I would expect that if we ever did make a reasonable basic income, 20% would be an oligarchic, enormous inequality to reward ourselves with. Change the heading to be more general about reinvesting some proportion of basic income money to us, and I'd rather that we at least have some caveat about how this is likely to change depending on the size of the UBI and the development costs of the currency, since 20% is a very arbitrary number.

File: CurrencyDesign.md (hunk 17 above)
Comment (markulrich) [reply CurrencyDesign.md]: Currently there is no long-term inflation mechanism.

The prevailing sentiment is that long-term inflation makes it's a very poor store of value, see https://twitter.com/VitalikButerin/status/980747076945379328, and I think the economic arguments for this are compelling. Perhaps more importantly, if enough people feel this is true, then that will decrease how long they are willing to hold the coin, which makes it true. We can wait and see what happens with Manna.

Perhaps a good goal would be for us would be to have a constitution and governance structure in place after the first 1000 days of distribution. At that point, we can establish how difficult it would be for the board to change things like inflation (ranging from "it's impossible, doing so would be a hard fork that is no longer Raha" to "we should always have some, and select it based on these factors").

File: CurrencyDesign.md (hunk 19 above)
Comment (markulrich) [reply CurrencyDesign.md]: Starting with a relatively large portion means we could accomplish much much more than rewarding maintainers if that is what members want.

It will be extremely important to safeguard against abuse of power, created #3 for us to (slowly over the next year) dump research and expand on the Governance model.

For phrasing what would be better? Perhaps something like

#### 20% of distribution to government, 80% to members
20% of the initial distribution and reclaimed moneys will be managed in a 100% transparent fashion by the Raha Government. Initially this 20% could be used for network development, growth, and maintenance. If Raha becomes successful enough, we could hold member sorties on what to do with excess funds such as returning over 80% to members or supporting education and healthcare initiatives!

File: CurrencyDesign.md (hunk 14 above)
Comment (markulrich) [reply CurrencyDesign.md]: Because the network values are superlinear, the value per user distributed each day will increase as the network grows. If when there are 1M users each coin is worth $1, then each person is getting 5M*.8/1M = 4 RAHA = $4/day. If next day there are 10M members and Metcalfe's law holds, then each coin now worth almost $100, so each person only gets .4 RAHA but that amounts to about $40/day.

It's very true that early adopters might be rewarded very well (IF they hold on to instead of selling their coins and IF they stay active and IF Raha network continues to grow), but they are the ones taking time to contribute to a community and believe in a vision when there is no pay-out in sight.

Created #4 to further discuss.

**Comment by rahulgi** [comment]
Further discussion can occur as more targeted issues and PR's!
